### PR TITLE
Revert old legacyFilesToAppend and vendorStaticStyles behavior as deprecated api

### LINKED ISSUE #5537: app.legacyFilesToAppend.push is not a function
After updating to Ember-cli 2.4.0 I get the following error:

"ActionView::Template::Error (:frontend has failed to build: app.legacyFilesToAppend.push is not a function)"

### PR DESCRIPTION
**Pull request by ro0gr**
Fixes #5537 

Moved handling of output files into new private props called `_styleOutputFiles` and `_scriptOutputFiles`. Saying the truth I'm totally unsure about naming here, would appreciate any suggestions :)

### CODE REVIEW

File: lib/broccoli/ember-app.js [hunk 1]
```diff
@@ -1136,9 +1136,20 @@ EmberApp.prototype.javascript = function() {
     annotation: 'Concat: App'
   });
 
+  if (this.legacyFilesToAppend.length > 0) {
+    deprecate('Usage of EmberApp.legacyFilesToAppend is deprecated. Please use EmberApp.import instead');
```
Comment (rwjblue) [inline lib/broccoli/ember-app.js]: Can you list the files to let them know what might have added them?

File: lib/broccoli/ember-app.js [hunk 2]
```diff
@@ -1194,9 +1204,18 @@ EmberApp.prototype.styles = function() {
 
   var preprocessedStyles = preprocessCss(stylesAndVendor, '/app/styles', '/assets', options);
 
+  if (this.vendorStaticStyles.length > 0) {
+    deprecate('Usage of EmberApp.vendorStaticStyles is deprecated. Please use EmberApp.import instead');
```
Comment (rwjblue) [inline lib/broccoli/ember-app.js]: Same as above, can you list the contents of `this.vendorStaticStyles`?

**Comment by rwjblue** [comment]
This looks like a good fix to me.  Basically, leave the old locations around and issue a deprecation if any files are present when we concat all the things.

Thanks for tackling this so quickly, if you can fix up my nit-picks here I'd like to land this in the 2.4.1 point release...

**Comment by ro0gr** [comment]
@rwjblue fixed according to your comments and rebased. now waiting for ci :eyes:

**Comment by rwjblue** [comment]
Thank you!!!

**Pull request merged**